src/main/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferService.java

```diff
@@ -77,6 +77,10 @@ public void updateConversationTransferStatus(UUID inboundConversationId, Convers
 
     public void updateConversationTransferStatusWithFailure(UUID inboundConversationId, String failureCode) {
         transferRepository.updateConversationStatusWithFailure(inboundConversationId, failureCode);
+
+        // remove the in-memory activity to end the conversation
+        activityService.removeConversationActivityTimestamp(inboundConversationId);
```

Comment:
Might be out of scope for this ticket, though myself and @MohammadIqbalAD-NHS (paired-review) agree that perhaps removing the `Timestamp` could enhance readability.

```suggestion
        activityService.removeConversationActivity(inboundConversationId);
```